CMakeLists.txt

```diff
@@ -1,95 +1,75 @@
 cmake_minimum_required(VERSION 3.15)
-project(bufr VERSION 11.3.0)
-set(${PROJECT_NAME}_VERSION ${PROJECT_VERSION} CACHE INTERNAL "${PROJECT_NAME} version number")
-enable_language (Fortran)
-
-if (NOT CMAKE_BUILD_TYPE)
-  set (CMAKE_BUILD_TYPE RELEASE CACHE STRING
-    "Choose the type of build, options are: PRODUCTION Debug Release."
-    FORCE)
+
+file(STRINGS "VERSION" pVersion)
+
+project(
+    bufr 
+    VERSION ${pVersion}
+    LANGUAGES C Fortran)
+
+if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release"
+                                               "MinSizeRel" "RelWithDebInfo")
 endif()
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+if(NOT CMAKE_C_COMPILER_ID MATCHES "^(Intel|GNU|Clang|AppleClang)$")
+  message(WARNING "Compiler not officially supported: ${CMAKE_C_COMPILER_ID}")
+endif()
+
+set(definitions "UNDERSCORE" "NFILES=32" "MAXCD=250" "MAXNC=600" "MXNAF=3")
+add_compile_definitions(definitions)
+
+if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
+  include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Intel.cmake)
   set(IntelComp true )
```

Comment:
please remove including these files. Explicitly set flags.  See other repos for how this is done to be consistent with other NCEPLIBS.